### ISSUE TITLE
A Minimal Span-Based Neural Constituency Parser

### ISSUE THREAD

**Issue by Tahkare** [Tahkare/PhD-archive#2]
## TL;DR
This paper presents a neural-based architecture for a span-based constituency parser. To sum it up briefly, the architecture computes a score for each span and also for each span/label combination. Then we can use a modified CYK recursion to compute the tree that will return the best score.

### Article Link
https://www.aclweb.org/anthology/P17-1076.pdf

### Author
M. Stern, J. Andreas & D. Klein

## Key Takeaways
* Architecture for a constituency parser
* Give an easy solution and discuss some potential alternatives for training and scoring

## Comments/ Questions
To compute the best score, we use the following equations :
* s<sub>label</sub>(i,j,l) is the score of the (i,j) span with label l
* s<sub>span</sub>(i,j) is the score of the span (i,j)
* s<sub>best</sub>(i,i+1) = max<sub>l</sub> [s<sub>label</sub>(i, i+1, l)] since we can't split a leaf
* s<sub>split</sub>(i,k,j) = s<sub>span</sub>(i,k) + s<sub>span</sub>(k,j) is the score of a chosen split
* s'<sub>split</sub>(i,k,j) = s<sub>split</sub>(i,k,j) + s<sub>best</sub>(i,k) + s<sub>best</sub>(k,j) is the score of a chosen split with the labels of the two sub-trees taken into account
And we end up with :
* s<sub>best</sub>(i,j) = max<sub>l</sub> [s<sub>label</sub>(i,j,l)] + max<sub>k</sub> [ s'<sub>split</sub>(i,k,j)] which is the best possible split with the best possible category.

We then just need to remember the splits and labels chosen to build our tree.
